# **Heart Failure Data: Extract, Transform and Load**

## Objectives

* Extract, clean and save the heart failure data.

## Inputs

* Dataset file data/raw/heart.csv

## Outputs

* A cleaned csv file, to be named later.

## Additional Comments

* Wasted at least 30 minutes trying to get into the correct working directory.  My tremor meant that I was clicking, using my trackpad, twice on the white Run icon to the left of the notebook cells and so traversing one too many directories.  Changing to selecting a cell and then running it with Shift + Enter solved the problem.





---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Code Institute - Data Analytics\\heart-failure-analysis\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Code Institute - Data Analytics\\heart-failure-analysis'

# Extract

In [4]:
import pandas as pd, numpy as np, matplotlib.pyplot as plt, seaborn as sns

In [6]:
df = pd.read_csv("data/raw/heart.csv")
df.head()

,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,ST_Slope,HeartDisease
0,40,M,ATA,140,289,0,Normal,172,N,0.0,Up,0
1,49,F,NAP,160,180,0,Normal,156,N,1.0,Flat,1
2,37,M,ATA,130,283,0,ST,98,N,0.0,Up,0
3,48,F,ASY,138,214,0,Normal,108,Y,1.5,Flat,1
4,54,M,NAP,150,195,0,Normal,122,N,0.0,Up,0


In [9]:
df.shape

(918, 12)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


In [13]:
df.isnull().sum()

Age               0
Sex               0
ChestPainType     0
RestingBP         0
Cholesterol       0
FastingBS         0
RestingECG        0
MaxHR             0
ExerciseAngina    0
Oldpeak           0
ST_Slope          0
HeartDisease      0
dtype: int64

In [15]:
df.duplicated().sum()

0

The investigation of the heart.csv file shows that it is comprised of 918 rows (labelled from 0 to 917), and 12 columns (labelled from 0 to 11).

The various columns of data can be divided into numeric and non-numeric datatypes (7 and 5 columns respectively).  An alternative categorisation may be made based on other features of the data such as demographics, clinical chemistry laboratory results, or physiological measures of cardiac function, emphasizing the richness of the dataset, despite its small size.

There appear to be no missing values or repeated rows of data.

In an idwal world I would like to see more information on the provenance of the data.  After using the Internet Archive it appears that one or more datasets from the 1990s were the source for the dataset but the current dataset makes no mention of how the current dataset was derived.  I am therefore sceptical with regard to its quality.

#   Column           Dtype        Category            Meaning
--  ------           -----        --------            -------
 0   Age             numeric      Demographics        Age (at diagnosis?)
 1   Sex             text         Demographics        Sex
 2   ChestPainType   text         Cardiac function    Type of chest pain
 3   RestingBP       numeric      Cardiac function    Resting blood pressure
 4   Cholesterol     numeric      Clinical chemistry  Cholesterol level
 5   FastingBS       numeric      Clinical chemistry  Fasting blood sugar
 6   RestingECG      text         Cardiac function    Resting electrocardiogram
 7   MaxHR           numeric      Cardiac function    Maximum heart rate
 8   ExerciseAngina  text         Cardiac function    Angina on exercising
 9   Oldpeak         numeric      Cardiac function    ECG ST segment level
 10  ST_Slope        text         Cardiac function    ST segment slope
 11  HeartDisease    numeric      Cardiac function    Presence of heart disease
Data types: numeric(7), text(5)

---

# Transform

---

NOTE

* You may add as many sections as you want, as long as it supports your project workflow.
* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)

---

# Push files to Repo

* In cases where you don't need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [16]:
import os
try:
  # create your folder here
  # os.makedirs(name='')
except Exception as e:
  print(e)


IndentationError: expected an indented block after 'try' statement on line 2 (553063055.py, line 5)